src/main/java/de/tum/in/www1/artemis/domain/quiz/DragAndDropQuestion.java

```diff
@@ -176,16 +162,21 @@ public void afterCreate() {
         }
     }
 
-    @PreUpdate
-    public void onUpdate() {
-        // move file and delete old file if necessary
-        backgroundFilePath = fileService.manageFilesForUpdatedFilePath(prevBackgroundFilePath, backgroundFilePath, FilePathService.getDragAndDropBackgroundFilePath(), getId());
-    }
-
+    /**
+     * This method is called when deleting the entity. It makes sure that the corresponding file is deleted as well.
+     */
     @PostRemove
     public void onDelete() {
         // delete old file if necessary
-        fileService.manageFilesForUpdatedFilePath(prevBackgroundFilePath, null, FilePathService.getDragAndDropBackgroundFilePath(), getId());
+        try {
+            if (backgroundFilePath != null) {
+                fileService.deleteFiles(List.of(Path.of(fileService.actualPathForPublicPathOrThrow(backgroundFilePath))));
+            }
+        }
+        catch (FilePathParsingException e) {
+            // if the file path is invalid, we don't need to delete it
+            log.warn("Could not delete file with path " + backgroundFilePath + ". Assume already deleted, entity can be removed.", e);
```

Comment:
You can still use placeholders, the logger automatically detects the additonal argument as an exception.
```suggestion
            log.warn("Could not delete file with path {}. Assume already deleted, entity can be removed.", backgroundFilePath, e);
```